libvast/src/system/configuration.cpp

```diff
@@ -229,6 +246,11 @@ caf::error configuration::parse(int argc, char** argv) {
   }
   // Flatten everything for simplicity.
   merged_config = flatten(merged_config);
+  // Overwrite current config with VAST_ environment variables, potentially
```

Comment:
It should be noted somewhere that this cannot work for all options, because some are parsed _before_ we execute this piece of code.

Reply:
I've fixed this up so that it applies to (almost) all options.

That said, I don't understand why setting `VAST_VERBOSITY` doesn't work. The logger initialization comes after the config parsing, what am I missing?

Reply:
Note that setting `VAST_CONSOLE_VERBOSITY=debug` works as expected.

I found this in `logger.cpp`:
```c++
  // Allow `vast.verbosity` from the command-line to overwrite
  // the `vast.console-verbosity` setting from the config file.
  auto verbosity = caf::get_if<std::string>(&cfg_cmd, "vast.verbosity");
  if (verbosity) {
    if (loglevel_to_int(*verbosity, -1) < 0) {
      fmt::print(stderr,
                 "failed to start logger; vast.verbosity '{}' is invalid\n",
                 *verbosity);
      return false;
    }
    console_verbosity = *verbosity;
  }
```

Having seen that, I'm inclined to change the command-line pre-processing from

```c++
  const auto replacements = std::vector<std::pair<std::string, std::string>>{
    {"-qqq", "--verbosity=quiet"}, {"-qq", "--verbosity=error"},
    {"-q", "--verbosity=warning"}, {"-v", "--verbosity=verbose"},
    {"-vv", "--verbosity=debug"},  {"-vvv", "--verbosity=trace"},
  };
```
to
```c++
  const auto replacements = std::vector<std::pair<std::string, std::string>>{
    {"-qqq", "--console-verbosity=quiet"}, {"-qq", "--console-verbosity=error"},
    {"-q", "--console-verbosity=warning"}, {"-v", "--console-verbosity=verbose"},
    {"-vv", "--console-verbosity=debug"},  {"-vvv", "--console-verbosity=trace"},
  };
```

Objections?

Reply:
None from my side, although that is a breaking change that needs to be documented separately.